lib/ddtrace/contrib/redis/patcher.rb

```diff
@@ -72,16 +73,28 @@ def call_pipeline(*args, &block)
             def datadog_pin
               @datadog_pin ||= begin
                 pin = Datadog::Pin.new(
-                  Datadog.configuration[:redis][:service_name],
+                  datadog_configuration[:service_name],
                   app: Ext::APP,
                   app_type: Datadog::Ext::AppTypes::DB,
                   tracer: Datadog.configuration[:redis][:tracer]
                 )
                 pin.onto(self)
               end
             end
+
+            private
+
+            def datadog_configuration
+              @datadog_configuration ||= Datadog.configuration[:redis, resolver.resolve(options)]
```

Comment:
I don't think it makes sense to use a custom instance of the resolver here; I think passing the `options` through should be sufficient, as it will use the resolver on the integration.

Also, I'd be hesitant to cache the `datadog_configuration` like this; although its unlikely that the configuration will change from trace to trace for this same `Client` instance, I'm worried about long-living objects like this will hold references to trace components and cause memory leaks.

It also makes sense that we wouldn't want to re-resolve the configuration multiple times per trace; I would suggest whatever references this store the result in a local variable to avoid multiple calls, but also avoid caching as well.

Reply:
I do not see any risk in caching the configuration for the same client instance.
I do not see any use case where you will use the same client to connect to different urls.